src/protocol/vm/state.rs

```diff
@@ -139,52 +135,63 @@ impl VMPoolState<PreCachedDB> {
         Ok(state)
     }
 
-    async fn set_engine(&mut self, adapter_contract_path: String) -> Result<(), SimulationError> {
+    async fn set_engine(&mut self) -> Result<(), SimulationError> {
```

Comment:
I still don't quite like this function though... I expect a `set_xxx` function to have a `xxx` as arg and set it on the struct. Here we are actually ensuring that the engine is set and creating one if not. I think something like `ensure_engine` would be more accurate. What do you think?

Also I think it would be nice to have a way to actually pass the engine, (that would help to fix slow tests) but probably out of scope for this PR

Reply:
Merging without this since I understand you will take care of it in your PR!